### PR TITLE
fix: support for iOS 14+ with `setAdvertiserTrackingEnabled`; remove rectangle banner on iOS

### PR DESCRIPTION
**Pull request by TMaszko**
### Summary
In this PR:
- [x] removed rectangle banner on ios
- [x] added support for ios 14+ by implementing `setAdvertiserTrackingEnabled` method
- [x] added readme for `setAdvertiserTrackingEnabled` method
### Test plan

1. Open iOS simulator/device with ios 14+
2. Give app tracking permission
3. Check all types of ads -> they should be visible

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -241,7 +241,7 @@ Banners are available in 3 sizes:
 
 - `standard` (BANNER_HEIGHT_50)
 - `large` (BANNER_HEIGHT_90)
-- `rectangle` (RECTANGLE_HEIGHT_250)
+- `rectangle` (RECTANGLE_HEIGHT_250: (currently only for Android)
```
Comment (thymikee) [inline README.md:244]:
```suggestion
- `rectangle` (RECTANGLE_HEIGHT_250) – currently Android only
```

**Review by mrousavy — APPROVED** [review]
Looks good to me - but doesn't the native android SDK also provide the personalization enable/disable func?

File: README.md [hunk 2]
```diff
@@ -384,8 +384,20 @@ Requests permission to track the user. Requires an [`NSUserTrackingUsageDescript
...
   AdSettings.setAdvertiserIDCollectionEnabled(true);
+  AdSettings.setAdvertiserTrackingEnabled(true);
+}
+```
+
+> Requires iOS 14. On Android and iOS versions below 14, this will always be no-op.
+
+### setAdvertiserTrackingEnabled
+
+Enables or disables ads tracking. Since the iOS 14 API has been introduced you will need set this flag in order to inform Facebook whether you should receive personalized ads. ( See [Advertising Tracking Enabled For Audience Network](https://developers.facebook.com/docs/audience-network/guides/advertising-tracking-enabled))
```
Comment (thymikee) [inline README.md:397]:
```suggestion
Enables or disables personalized ads tracking on iOS 14+. See Facebook docs on [Advertising Tracking Enabled For Audience Network](https://developers.facebook.com/docs/audience-network/guides/advertising-tracking-enabled)
```

File: README.md [hunk 3]
```diff
@@ -384,8 +384,20 @@ Requests permission to track the user. Requires an [`NSUserTrackingUsageDescript
 
 ```js
 const trackingStatus = await AdSettings.requestTrackingPermission();
-if (trackingStatus === 'authorized' || trackingStatus === 'unavailable')
+if (trackingStatus === 'authorized' || trackingStatus === 'unavailable') {
   AdSettings.setAdvertiserIDCollectionEnabled(true);
+  AdSettings.setAdvertiserTrackingEnabled(true);
```
Comment (thymikee) [inline README.md:389]: this is unrelated, let's remove

File: README.md [hunk 4]
```diff
@@ -384,8 +384,20 @@ Requests permission to track the user. Requires an [`NSUserTrackingUsageDescript
...
+}
+```
+
+> Requires iOS 14. On Android and iOS versions below 14, this will always be no-op.
+
+### setAdvertiserTrackingEnabled
```
Comment (thymikee) [inline README.md:395]: Let's add a clear notice that FB won't display adds when this is not set to true

**Comment by TMaszko** [comment]
@mrousavy hmm I did this upgrade according to this guide https://developers.facebook.com/docs/audience-network/guides/advertising-tracking-enabled. I haven't seen anything about Android :(.

File: README.md [hunk 5]
```diff
@@ -390,13 +390,13 @@ if (trackingStatus === 'authorized' || trackingStatus === 'unavailable') {
...
 
-> Requires iOS 14. On Android and iOS versions below 14, this will always be no-op.
-> **Note**: FB won't display adds when this is not set to true.
-
 ### setAdvertiserTrackingEnabled
 
 Enables or disables personalized ads tracking on iOS 14+. See Facebook docs on [Advertising Tracking Enabled For Audience Network](https://developers.facebook.com/docs/audience-network/guides/advertising-tracking-enabled)
 
+> Requires iOS 14. On Android and iOS versions below 14, this will always be no-op.
+> **Note**: FB won't display adds when this is not set to true.
```
Comment (thymikee) [inline README.md:398]:
```suggestion
> **Important: FB won't display adds until this is set to `true`.**
```

**Review by thymikee — APPROVED** [review]

**Comment by bliang** [comment]
How were you able to get around the requirement for:

```
[!] CocoaPods could not find compatible versions for pod "FBSDKShareKit":
  In snapshot (Podfile.lock):
    FBSDKShareKit (= 5.8, = 5.8.0)

  In Podfile:
    react-native-fbsdk (from `../node_modules/react-native-fbsdk`) was resolved to 3.0.0, which depends on
      react-native-fbsdk/Share (= 3.0.0) was resolved to 3.0.0, which depends on
        FBSDKShareKit (~> 8.1)


You have either:
 * changed the constraints of dependency `FBSDKShareKit` inside your development pod `react-native-fbsdk`.
   You should run `pod update FBSDKShareKit` to apply changes you've made.
```

**Comment by mrousavy** [comment]
Did you run what the command output suggested?

```
pod repo update
pod update FBSDKShareKit
```

**Comment by bliang** [comment]
@mrousavy I have.

**Comment by bliang** [comment]
Looks like Podfile.lock needed to be cleared first since it was pinned to a previous version. Thanks!
